dbms/src/Storages/S3/Credentials.cpp

```
@@ -218,6 +224,207 @@ void STSAssumeRoleWebIdentityCredentialsProvider::refreshIfExpired()
+    Aws::Map<Aws::String, Aws::String> params;
+    params["Action"] = "AssumeRoleWithOIDC";
+    params["Format"] = "JSON";
+    params["Version"] = "2015-04-01";
+    params["RoleSessionName"] = m_session_name;
+    params["OIDCProviderArn"] = m_oidc_provider_arn;
+    params["RoleArn"] = m_role_arn;
+    params["OIDCToken"] = m_token;
+    params["Timestamp"] = Aws::Utils::DateTime::Now().ToGmtString(Aws::Utils::DateFormat::ISO_8601);
+
+    // build http query_string from params
+    FmtBuffer buff;
```

Clang (14.0.6)

[error] variable 'buff' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)